From: Knud Baastrup <kba@deif.com>
Date: Sun, 10 Aug 2014 22:07:39 +0200

Use call back functions.

Ensures that we use the call back functions (with io lock support) when
using the IOCTL interface. Fixed by Johnny Rasmussen <jra@deif.com>.

diff --git a/master/ioctl.c b/master/ioctl.c
--- a/master/ioctl.c
+++ b/master/ioctl.c
@@ -2048,7 +2048,11 @@ static ATTRIBUTES int ec_ioctl_send(
     if (ec_ioctl_lock_down_interruptible(&master->master_sem))
         return -EINTR;
 
-    sent_bytes = ecrt_master_send(master);
+    if (master->send_cb != NULL) {
+        master->send_cb(master->cb_data);
+        sent_bytes = 0;
+    } else
+        sent_bytes = ecrt_master_send(master);
 
     ec_ioctl_lock_up(&master->master_sem);
 
@@ -2080,7 +2084,10 @@ static ATTRIBUTES int ec_ioctl_receive(
     if (ec_ioctl_lock_down_interruptible(&master->master_sem))
         return -EINTR;
 
-    ecrt_master_receive(master);
+    if (master->receive_cb != NULL)
+        master->receive_cb(master->cb_data);
+    else
+        ecrt_master_receive(master);
 
     ec_ioctl_lock_up(&master->master_sem);
     return 0;
